wordpress-mobile/WordPress-Android
# My Site Dashboard: Tabs - Add "Site Menu" QS step, show QS focus point on tab

Parent #16006

This PR adds `Site Menu` QS step when needed, displays QS focus point on the tab, and requests pending QS task on tab change. 
The "Site Menu" QS step is needed when QS task starts from `Dashboard`/ `Home` tab but the corresponding menu item is present on the `Site Menu` tab. In this case, an intermediate step is introduced that instructs users to tap `Site Menu` tab to continue.  

Video captures:

- "Site Menu" QS step (timstamp 0:23)
- Auto-scroll (on continue from "Site Menu" QS step (timstamp 0:26)
- FAB (timstamp 0:38)
- Integration from other parts of the app (timestamp 0.13 - Homepage, 0.31 - Publicize, 0.47 - Reader)

https://user-images.githubusercontent.com/1405144/159706819-c79d6d8e-2087-48ed-bf6b-ed679ad96a7c.mp4

To test:

**Test.1: Regression test Quick Start in single view with tabs disabled**

1. Launch the app.
2. Go to `My Site` -> `Me` -> `App Settings` -> `Debug settings`.
3. Turn off the `MySiteDashboardTabsFeatureConfig` flag under the `Features in development` section.
4. Restart the app.
5. Logout/login and select "Show me around" when the quick start prompt is shown.
6. Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt).

**Test.2: Quick Start with tabs enabled**
1. Launch the app.
2. Go to `My Site` -> `Me` -> `App Settings` -> `Debug settings`.
3. Turn on the `MySiteDashboardTabsFeatureConfig` flag under the `Features in development` section.
4. Restart the app.
5. Logout/login and select "Show me around" when the quick start prompt is shown.
6. Note the "Quick Start" shows in the `Dashboard`/ `Home` tab only. 
7. Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt).
[ Notice that in `Test 1 - Quick start notice display` in the text file, when quick start task is available only on `Site Menu` tab, `Site Menu` tab shows QS focus point. On clicking the tab, the focus point should disappear and the appropriate site menu item should get QS focus point.]

**Review Instructions**

Code review from only one reviewer is sufficient.

## Regression Notes
1. Potential unintended areas of impact
Quick start actions/tasks don't work as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tested

3. What automated tests I added (or what prevented me from doing so)
Unit tests were updated to reflect refactoring efforts

PR submission checklist:
